src/Dependencies/PooledObjects/ArrayBuilder.cs

```diff
@@ -720,5 +720,24 @@ public ImmutableArray<S> SelectDistinct<S>(Func<T, S> selector)
             set.Free();
             return result.ToImmutableAndFree();
         }
+
+        public static PooledDisposer<ArrayBuilder<T>> GetInstance(out ArrayBuilder<T> instance)
```

Comment:
Definitely need compiler sign off here. This will maje some unusable members show up in completion for a very common type.

Reply:
Why is this an unusable member?

Reply:
Perhaps instead of banning the API, we can define `ROSLYN_COMPILER` symbol and wrap these in `#if !ROSLYN_COMPILER`.

Reply:
Wrapping in `#if !COMPILERCORE` seems preferrable to me over adding the APIs to the compiler and then banning them, so the APIs cannot be accidentally referenced.

Reply:
I'll add a new symbol `MICROSOFT_CODEANALYSIS_POOLEDOBJECTS_NO_POOLED_DISPOSER` and define it in compiler projects.